File: oop.js
class Persona {
    constructor(nombre = "", apellido = ""){
        //console.log("Instanciando la clase Persona")
        this._nombre = nombre;
        this._apellido = apellido
    }

    quienSoy(){
        return `${this._nombre} ${this._apellido}`
    }
}

let persona = new Persona() 
console.log(persona.quienSoy())

let persona2 = new Persona("John", "Doe") 
console.log(persona2.quienSoy())


class Estudiante extends Persona {
    constructor(nombre, apellido, facultad){
        super(nombre, apellido)
        this._facultad = facultad
        //console.log("Instanciando la clase Estudiante")
    }

    quienSoy(){
        return `Soy ${this._nombre} ${this._apellido} de la facultad de: ${this._facultad}`
    }
}

let estudiante = new Estudiante("Jane", "Doe", "Medicina")
console.log(estudiante.quienSoy())

File: oop.py
class Persona:
    def __init__(self, nombre = "", apellido = ""):
        print("Instanciando la clase Persona")
        self.nombre = nombre
        self.apellido = apellido

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_nombre(self):
        return self.nombre

    def quien_soy(self):
        return f"{self.nombre} {self.apellido}"


persona = Persona()
print(persona.quien_soy())

persona2 = Persona("John", "Doe")
print(persona2.quien_soy())

class Estudiante(Persona):
    def __init__(self, nombre, apellido, facultad):
        super().__init__(nombre, apellido)
        self.__facultad = facultad

    def quien_soy(self):
        return f"Soy {self.nombre} {self.apellido} de la facultad de: {self.__facultad}"


estudiante = Estudiante("Jane", "Doe", "Medicina")
print(estudiante.quien_soy())